pytorch_lightning/plugins/environments/lsf_environment.py

```diff
@@ -0,0 +1,160 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import socket
+
+from pytorch_lightning import _logger as log
+from pytorch_lightning.plugins.environments import ClusterEnvironment
+
+
+class LSFEnvironment(ClusterEnvironment):
+    """
+    An environment for running on clusters managed by the LSF resource manager.
+
+    It is expected that any execution using this ClusterEnvironment was executed
+    using the Job Step Manager i.e. ``jsrun``.
+
+    This plugin expects the following environment variables.
+
+    LSB_JOBID:
+        The LSF assigned job ID
+
+    LSB_HOSTS:
+        The hosts used in the job. This string is expected to have the format "batch <rank_0_host> ...."
+
+    JSM_NAMESPACE_LOCAL_RANK:
+        The node local rank for the task. This environment variable is set by jsrun
+
+    JSM_NAMESPACE_SIZE:
+        The world size for the task. This environment variable is set by jsrun
+    """
+
+    def __init__(self):
+        self._master_address = self._get_master_address()
+        self._master_port = self._get_master_port()
+        log.debug(f"MASTER_ADDR: {self._master_address}")
+        log.debug(f"MASTER_PORT: {self._master_port}")
+
+    @staticmethod
+    def is_using_lsf() -> bool:
+        """ Returns ``True`` if the current process was launched using the jsrun command. """
+        required_env_vars = (
+            "LSB_JOBID",
+            "LSB_HOSTS",
+            "JSM_NAMESPACE_LOCAL_RANK",
+            "JSM_NAMESPACE_SIZE",
+        )
+        return all(v in os.environ for v in required_env_vars)
+
+    def creates_children(self) -> bool:
+        return True
+
+    def master_address(self):
+        """ The master address is read from a list of hosts contained in the environment variable `LSB_HOSTS`. """
+        return self._master_address
+
+    def master_port(self):
+        """ THe master port gets calculated from the LSF job ID. """
+        return self._master_port
+
+    def world_size(self):
+        """ The world size is read from the environment variable `JSM_NAMESPACE_SIZE`. """
+        var = "JSM_NAMESPACE_SIZE"
```

Comment:
Same here ? To avoid possible typos ?

Reply:
yes, this would apply to all environments because it's part of the base interface. 
This todo is tracked here: #6303
Could be a good issue for a new contributor.